### PR TITLE
Remove api_create from R & ggplot2 examples

### PR DESCRIPTION
**Pull request by CanerIrfanoglu**
This branch removes all instances of `api_create` from the examples in `_posts/r/`  and `_posts/ggplot2/`. It contains new `RdepLibs.R` file to install the dependency packages.

## Main Notes 
1. `RdepLibs.R` file is created in the root directory of the repo. The file contains all the required R libraries with the path of the example where the dependency is first required.

2. Removed setting plotly credentials for `knitr::opts_chunk$set` in all the examples

_Before_
```{r, echo = FALSE, message=FALSE}
knitr::opts_chunk$set(message = FALSE, warning=FALSE)
Sys.setenv("plotly_username"="RPlotBot")
Sys.setenv("plotly_api_key"="q0lz6r5efr")
```

_After_
```{r, echo = FALSE, message=FALSE}
knitr::opts_chunk$set(message = FALSE, warning=FALSE)
```

3. Removed the `api_create` calls & `results = 'hide'` setting from all the examples. 

_Example Before_
```{r}
```{r, results = 'hide'}
library(plotly)

p <- plot_ly(data = iris, x = ~Sepal.Length, y = ~Petal.Length)

# Create a shareable link to your chart
# Set up API credentials: https://plot.ly/r/getting-started
chart_link = api_create(p, filename="scatter-basic")
chart_link
```
```{r}
```{r, echo=FALSE}
chart_link
```

_Example After_
```{r}
```{r}
library(plotly)

p <- plot_ly(data = iris, x = ~Sepal.Length, y = ~Petal.Length)

p
```

## Additional Notes

1. `_posts/r/Fundamentals` section contains documentation that would  possibly be dedocumented later. There are 6 Rmd files. Made edits only to `2015-08-10-knitr.Rmd` since it had `knitr` related documentation not `api_create`.

2. `_posts/ggplot2/2015-08-10-getting-started_ggplot2_index.Rmd` — Initialization for online plotting describes `api_create` might be dedocumented.

3. Added following to all examples utilizing `mapbox`.

```{r, echo = FALSE}
# Mapbox token for plotly | this one is for plot_mapbox figure
mapboxToken <- paste("pk.eyJ1IjoicGxvdGx5bWFwYm94IiwiYSI6ImNqdnBvNDMyaT",
                "AxYzkzeW5ubWdpZ2VjbmMifQ.TXcBE-xg9BFdV2ocecc_7g", sep = "")

# Setting mapbox token for R environment 
Sys.setenv("MAPBOX_TOKEN" = mapboxToken)
``` 

4. `_posts/r/financial/2019-09-17-funnel-charts` required installing plotly from github as indicated in https://github.com/ropensci/plotly/issues/1623. 

5. `_posts/ggplot2/2016-11-29-facet-grid/Geom Line` within the example there was a `corr_plot`(https://plot.ly/ggplot2/facet_grid/) throwing error and it seemed irrelevant to the example. Removed it.

@nicolaskruchten @rpkyle

### CODE REVIEW

File: _posts/r/basic/2015-07-30-line-and-scatter.Rmd [hunk 1]
```diff
@@ -39,24 +41,17 @@ packageVersion('plotly')
 
 ### Basic Scatter Plot
 
-```{r, results = 'hide'}
```
Comment (nicolaskruchten) [inline _posts/r/basic/2015-07-30-line-and-scatter.Rmd]: I wonder why this pattern was used before? the first cell uses `results = 'hide'` but the subsequent one has ostensibly the same output (same final line) without `results = 'hide'`... Was this to suppress some warnings or undesirable output? 

If not, or if nothing obvious comes to mind, this is a fine change to me as far as I'm concerned

File: _posts/r/basic/2015-07-30-line-and-scatter.Rmd [hunk 2]
```diff
@@ -16,9 +16,11 @@ output:
 ---
 
 ```{r, echo = FALSE, message=FALSE}
-knitr::opts_chunk$set(message = FALSE, warning=FALSE)
-Sys.setenv("plotly_username"="RPlotBot")
-Sys.setenv("plotly_api_key"="q0lz6r5efr")
+knitr::opts_chunk$set(
+  message = FALSE,
+  warning = FALSE,
+  fig.width = 8.33,  # ~800px
```
Comment (nicolaskruchten) [inline _posts/r/basic/2015-07-30-line-and-scatter.Rmd]: these settings lead to a figure that's a bit too big actually, let's shrink them by about 25% in both dimensions?

**Comment by nicolaskruchten** [comment]
Looks great to me! Let's move forward :)

**Comment by nicolaskruchten** [comment]
OK nice, these all seem like pretty straightforward changes... no weird ones that don't run at all?

**Comment by CanerIrfanoglu** [comment]
> OK nice, these all seem like pretty straightforward changes... no weird ones that don't run at all?

Yes, I identified some already. Decided to run a command for the changes first. Then, run all examples individually, fix broken ones and note down the dependencies.

**Comment by nicolaskruchten** [comment]
How are things looking with this project? Nearing completion or...? :)

**Comment by CanerIrfanoglu** [comment]
> How are things looking with this project? Nearing completion or...? :)

Things are going pretty well with it. I spend some time to get _r/3d/3d-tri-surf/Maps_ example working. That is solved thanks to @mdsumner. 

The project is close to completion. I just need to run the remaining ggplot2 examples and document my changes.

**Comment by nicolaskruchten** [comment]
Thank you very much @CanerIrfanoglu for this PR, it's hitting prod today (although it won't be merged into this repo, as we moved all these files into https://github.com/plotly/plotly.r-docs)

Your help was invaluable in getting this project done, and in getting all these docs to run on every commit in CircleCI. Thanks again and have a great end-of-year!

cc @rpkyle

**Comment by CanerIrfanoglu** [comment]
@nicolaskruchten thanks for the update. Very happy to hear that my work was useful. Wishing you a great time with family and friends for the end of the year :)

**Comment by nicolaskruchten** [comment]
@Mahdis-z so this is the branch we used to build https://github.com/plotly/plotly.r-docs ... any commits you merged into the R or ggplot2 docs after this branch forked off of `source-design-merge` have been lost and will need to be back-ported... sorry!

**Comment by nicolaskruchten** [comment]
@Mahdis-z I think it's just this PR that needs to be resubmitted to the new repo: https://github.com/plotly/documentation/pull/1545/files

**Comment by jdamiba** [comment]
This issue has been addressed and these files now live in the plotly.r-docs repo. 

Great job @CanerIrfanoglu!

**Pull request closed without merging**